viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not (title and genre and rating):
+        return None
+    else:
+        movie = { "title" : title,
+                "genre" : genre,
+                "rating" : rating
+                }
+        return movie
+    
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    if not user_data["watched"]:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_list.append(movie["rating"])
+    average_rating = sum(rating_list) / len(rating_list)
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_list = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+    
+    most_watched_genre_frequency = 0
+    most_watched_genre = genre_list[0]
+    for genre in genre_list:
+        genre_frequency = genre_list.count(genre)

```

Comment:
Consider the fact that the time complexity of `count` is linear and it's nested in a for loop which brings us to O(n^2) time complexity. 

What if you built up a frequency dictionary using user_data["watched"] where the key was the genre name and the value was the count. Then you could loop through the dictionary to find the highest value.